coursemapper-kg/recommendation/app/services/course_materials/recommendation/youtube_service.py

```
@@ -42,17 +43,67 @@
+            try:
+                youtube = googleapiclient.discovery.build(api_service_name, api_version, developerKey=key)
+                request = youtube.search().list(
+                    part="snippet",
+                    maxResults=top_n,
+                    type="video",
+                    q=query,
+                    relevanceLanguage="en",
+                )
+                return request.execute(), youtube
+
+            except (ConnectionAbortedError, ConnectionResetError, timeout) as e:
```

Ruff (0.8.2)

`timeout` may be undefined, or defined from star imports

(F405)